lib/icon_data.dart

```diff
@@ -28,3 +28,13 @@ class IconDataRegular extends IconData {
           fontPackage: 'font_awesome_flutter',
         );
 }
+
+// Only available when Light font is installed
+class IconDataLight extends IconData {
```

Comment:
I agree that this might not be simplest, however it allows for easiest consumption for everyone.

If we really want to split the API for Pro/non-Pro, maybe let's just create a Pro version of this package with only the Pro icons.

Reply:
We don't want to split the api, that's the point. We cannot provide a package with the pro icons, because the data needed (the pro `icons.json`) has to be paid for and cannot be available for free in open source projects.